### PR TITLE
fix: Refactor Realtime Channel

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?

In preparation to add Authorization, we're separating concerns out of the module RealtimeChannel which had most of the logic within the same module.

This will make it easier to extend the Broadcast and Presence features.

We're also moving the Channels endpoints to a v3 route

### CODE REVIEW

File: lib/realtime_web/channels/realtime_channel.ex [hunk 1]
```diff
@@ -474,8 +412,7 @@ defmodule RealtimeWeb.RealtimeChannel do
   end
 
   defp presence_key(params) do
-    with key when is_binary(key) <- params["config"]["presence"]["key"],
-         true <- String.length(key) > 0 do
+    with key when is_binary(key) and key != "" <- params["config"]["presence"]["key"] do
```
Comment (philss) [inline lib/realtime_web/channels/realtime_channel.ex:415]: Nice :)

File: lib/realtime_web/channels/realtime_channel.ex [hunk 2]
```diff
@@ -502,7 +439,9 @@ defmodule RealtimeWeb.RealtimeChannel do
   end
 
   defp confirm_token(%{assigns: %{jwt_secret: jwt_secret, access_token: access_token} = assigns}) do
-    with jwt_secret_dec <- decrypt_jwt_secret(jwt_secret),
+    secure_key = Application.get_env(:realtime, :db_enc_key)
```
Comment (philss) [inline lib/realtime_web/channels/realtime_channel.ex:442]: Nitpick: wouldn't be better to raise if the secure key is not configured?

```suggestion
    secure_key = Application.fetch_env!(:realtime, :db_enc_key)
```

**Review by philss — COMMENTED** [review]
Nice refactor! LGTM :)

**Review by abc3 — APPROVED** [review]

File: lib/realtime_web/channels/realtime_channel.ex (hunk 2 above)
Comment (filipecabaco) [reply lib/realtime_web/channels/realtime_channel.ex:442]: oh great call! 🙏

**Pull request merged**